File: PayPark/app/src/main/java/com/example/paypark/PlateNumberValidation.kt
package com.example.paypark

class PlateNumberValidation {

    val regions = listOf(
        "B", "BZ", "IL", "BR", "IF", "BV", "HR", "CV", "CJ", "AB",
        "TM", "BH", "HD", "CS", "GR", "CL", "TL", "CT", "VN", "GL",
        "BC", "VS", "NT", "IS", "SV", "BT", "MS", "SM", "DJ", "GJ",
        "SB", "VL", "OT", "PH"
    )

    fun validateNumber(plateNumber: String): Boolean {
        val splitPlateNumber = plateNumber.split(" ")
        if (splitPlateNumber.size == 4) {
            if (splitPlateNumber[0] == "RO") {
                if (splitPlateNumber[1] in regions) {
                    if (splitPlateNumber[2].isDigitsOnly() && splitPlateNumber[3].isAlphaOnly() && splitPlateNumber[3].length == 3) {
                        return true
                    }
                }
            }
        }
        return false
    }

    // Extension function to check if a string contains only digits
    fun String.isDigitsOnly(): Boolean {
        return this.all { it.isDigit() }
    }

    // Extension function to check if a string contains only alphabetic characters
    fun String.isAlphaOnly(): Boolean {
        return this.all { it.isLetter() }
    }

}

File: PaymentManager/plateNumberValidation.py
def validate_number(plate_number, VALID_ROMANIAN_REGIONS):
    split_plate_number = plate_number.split()
    if len(split_plate_number) == 4:
        if split_plate_number[0] == "RO":
            if split_plate_number[1] in VALID_ROMANIAN_REGIONS:
                if split_plate_number[2].isdigit() and (split_plate_number[3].isalpha() and len(split_plate_number[3]) == 3):
                    return True
                else:
                    return False


def read_valid_regions():
    with open("resources/valid regions/valid_romanian_regions.txt", "r") as file:
        VALID_ROMANIAN_REGIONS = [line.strip() for line in file]

    return VALID_ROMANIAN_REGIONS
